### PR TITLE
Surface elevation shadow colour on Material

### PR DESCRIPTION
**Pull request by rock3r**
This PR surfaces the previously hardcoded colour for the elevation shadows to `Material` (where it defaults to the previous value of `0xFF000000`). This way one can create a widget that has a tinted shadow:

fuch-siaaaaaa

For example, one could use it as such:

```dart
class _FancyShades extends StatefulWidget {

  @override
  State<StatefulWidget> createState() => new _FancyShadesState();
}

class _FancyShadesState extends State<_FancyShades> {

  bool _isFuchsia = false;

  @override
  Widget build(BuildContext context) =>
      new GestureDetector(
        child: new Material(
          type: MaterialType.card,
          borderRadius: new BorderRadius.circular(2.0),
          elevation: _isFuchsia ? 10.0 : 2.0,
          shadowColor: _isFuchsia ? const Color(0xffff00ff) : Colors.black,
        ),
        onTap: () => setState(() => _isFuchsia = !_isFuchsia),
      );
}
```

Paired with @takhion

### CODE REVIEW

File: packages/flutter/lib/src/material/material.dart
Comment (rock3r) [inline packages/flutter/lib/src/material/material.dart]: We're not sure why this doesn't have a default value, is that correct?

File: packages/flutter/lib/src/material/material.dart
Comment (rock3r) [inline packages/flutter/lib/src/material/material.dart]: `color` seems to be missing an assert here?

File: packages/flutter/lib/src/widgets/implicit_animations.dart
Comment (rock3r) [inline packages/flutter/lib/src/widgets/implicit_animations.dart]: We're not asserting for not null for `animateShadowColor` because `animateColor` doesn't have one, but maybe they should both be asserting?

**Comment by yjbanov** [comment]
Thanks for the contribution!

LGTM, except I'm unsure if we want to animate shadow color by default. I'll check with someone on Monday before merging.

File: packages/flutter/lib/src/material/material.dart
Comment (yjbanov) [reply packages/flutter/lib/src/material/material.dart]: We're not animating color in `Material`, so `null` might be a valid value, although I haven't checked myself.

File: packages/flutter/lib/src/material/material.dart
Comment (yjbanov) [reply packages/flutter/lib/src/material/material.dart]: Yeah, I think it should be `0.0`.

File: packages/flutter/lib/src/widgets/implicit_animations.dart
Comment (yjbanov) [reply packages/flutter/lib/src/widgets/implicit_animations.dart]: Good point. Let's assert. `true` or `false` is all we need for these two.

File: packages/flutter/lib/src/material/material.dart
Comment (rock3r) [reply packages/flutter/lib/src/material/material.dart]: Yeah, we don't want to, but not animating it doesn't mean it could be null — material has a colour, be it white (default) or some other colour. I'm not sure what a null colour would represent for a material, semantically?

File: packages/flutter/lib/src/material/material.dart
Comment (rock3r) [reply packages/flutter/lib/src/material/material.dart]: dba7e0ced5eadda5bdf00a072ae2bf9f06517627

File: packages/flutter/lib/src/widgets/implicit_animations.dart
Comment (rock3r) [reply packages/flutter/lib/src/widgets/implicit_animations.dart]: dc9759f0ead8e18a50544fea4f5edb8f5e2d4248

File: packages/flutter/lib/src/material/material.dart
Comment (yjbanov) [reply packages/flutter/lib/src/material/material.dart]: Found it. Look for `_getBackgroundColor` in this file. `null` is a valid value for `color`, and means get the color from `ThemeData` based on the material type. We might do the same for `shadowColor` eventually, but for now let's just `assert`.

**Review by yjbanov — APPROVED** [review]

**Comment by yjbanov** [comment]
lgtm

**Comment by yjbanov** [comment]
Sorry, I forgot to ask about a test. There's a test titled 'Shadows animate smoothly'. Could you please create a similar test titled 'Shadow color animates smoothly'?

**Comment by rock3r** [comment]
Sure, should be able to do it before boarding my flight :)

**Comment by rock3r** [comment]
@yjbanov test added! I didn't find any assertion for `Color` that could work (maybe didn't look well enough) so I'm asserting per-component of the colours. Let me know if there's anything else I should use.

**Review by yjbanov — APPROVED** [review]

**Comment by yjbanov** [comment]
Yeah, we don't have anything to assert that two colors are close to each other. Let's land this as is and maybe I'll create a utility for that later. Thanks for staying through the review!

**Pull request merged**
